### PR TITLE
Add integration tests for tracing simulation

### PR DESCRIPTION
**Pull request by OrpazFishel**

### CODE REVIEW

File: src/test/java/com/google/research/bleth/simulator/TracingSimulationIT.java [hunk 1]
```diff
@@ -13,65 +16,159 @@
...
+                .setTransmissionThresholdRadius(transmissionRadius)
+                .setBeaconMovementStrategy(MOVE_UP)
+                .setObserverMovementStrategy(STATIONARY)
+                .setAwakenessCycle(AWAKENESS_CYCLE_EQUALS_TWO)
+                .setAwakenessDuration(AWAKENESS_DURATION_EQUALS_ONE)
+                .setAwakenessStrategyType(FIXED_AWAKENESS_STRATEGY_TYPE)
+                .build();
+
+        Map<IAgent, Location> initialAgentsToLocations = mapAgentsToLocationsOnBoard(simulation.getBoard());
+        Set<IAgent> observers = initialAgentsToLocations.keySet().stream().filter(agent -> agent instanceof Observer).collect(Collectors.toSet());
```
Comment (omermadmon) [inline src/test/java/com/google/research/bleth/simulator/TracingSimulationIT.java:113]: Why not just 
```java 
Set<IAgent> observers = simulation.observers.stream().collect(Collectors.toSet()); 
```
?

File: src/test/java/com/google/research/bleth/simulator/TracingSimulationIT.java [hunk 2]
```diff
@@ -13,65 +16,159 @@
...
+        Location observerInitialLocation = averageLocationOfObservers(observers);
```
Comment (omermadmon) [inline src/test/java/com/google/research/bleth/simulator/TracingSimulationIT.java:114]: We have a single observer here, why do we compute the average location?
(If the reason is to fit the pattern of the other tests, that's fine, just wanted to make sure).

**Review by omermadmon — COMMENTED** [review]
Looks good to me.
In general, we want to have these four levels of test classes:
- Builder tests - to verify the builders' verification logic works properly in edge-cases (already exists).
- Database-only tests - involving those classes which interact with the database, but not involving the `Simulation` class (these tests exist at #8).
- Simulation-only tests - involving the `Simulation` class, but not involving the db-related classes (these are the tests you have just wrote).
- End-to-end tests - involving both the simulation and the database interaction. Those tests will be written as soon as the db-related classes are merged to the master.

File: src/test/java/com/google/research/bleth/simulator/TracingSimulationIT.java (hunk 1 above)
Comment (OrpazFishel) [reply src/test/java/com/google/research/bleth/simulator/TracingSimulationIT.java]: I will change if simulation.observers remains protected and not private.
(Then simulation.observers will be good enough)

**Comment by OrpazFishel** [comment]
Refactor using BoardState.
